Add support for the `when` key in step phases (#3340)
teemtee/tmt: docs/releases.rst

support for :ref:`system.model-name</spec/hardware/system>`
hardware requirement.

> It is easier limit when step configuration is run by
> using :ref:`when-config` key.

Reviewer: ```suggestion
It is easier to limit a step execution using
the :ref:when-config key.
```
